express-mutations/server.ts

```diff
@@ -0,0 +1,102 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const sql = `
+                insert into "actors" ("firstName", "lastName")
+                values ($1, $2)
+                returning *
+                `;
+    const { firstName, lastName } = req.body;
+    if (!firstName) throw new ClientError(400, 'firstName is required');
+    if (!lastName) throw new ClientError(400, 'lastName is required');
+    const params = [firstName, lastName];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    res.status(201).json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (actorId === undefined)
+      throw new ClientError(400, 'actorId is required');
+    if (!Number.isInteger(+actorId))
+      throw new ClientError(400, 'actorId must be an integer');
+    const sql = `
+                update "actors"
+                set "firstName" = $1, "lastName" = $2
+                where "actorId" = $3
+                returning *;
+                `;
+    const { firstName, lastName } = req.body;
+    if (!firstName) throw new ClientError(400, 'firstName is required');
+    if (!lastName) throw new ClientError(400, 'lastName is required');
+    const params = [firstName, lastName, actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.status(200).json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.delete('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId))
```

Comment:
This check is a little different than the one you used in `put`. They both work, but it's good to be consistent in your code (it's easier to read and understand and to check that you haven't made any mistakes). I like this check better than the one in `put`, but both work.